apps/openmw/mwrender/renderingmanager.cpp

```diff
@@ -1195,6 +1195,10 @@ namespace MWRender
 
         mUniformNear->set(mNearClip);
         mUniformFar->set(mViewDistance);
+
+        // Since our fog is not radial yet, we should take FOV in account, otherwise terrain near viewing distance may disappear.
+        float distanceMult = std::cos(osg::DegreesToRadians(mFieldOfView)/2.f);
+        mTerrain->setViewDistance(mViewDistance * (distanceMult ? 1.f/distanceMult : 1.f));
```

Comment:
As something I haven't thought about, on extremely high FOVs (beyond 180 degrees) the mult becomes negative and bad things happen. Either distant mult should be absolute, or be replaced with a cosine of (pi-mFieldOfView/2.f) before usage, or there simply should be a check that distance mult is positive.

Reply:
me: "that doesn't work though!"

me: test it out ingame

me: https://i.imgur.com/HJooPAl.jpg

Reply:
Yeah. The problem is that with these changes the terrain disappears too. My bad, I wrote this part.
![screenshot021](https://user-images.githubusercontent.com/21265616/53685335-37775900-3d2a-11e9-9999-5ddc1eb3dedf.png)

Reply:
And which cap we should use for FOV? 110, as in UI?

Reply:
please no, there are valid reasons to play with unreasonably high FoVs, making the config file cap 179 or something just to keep it from being a broken value is fine

should also enforce a minimum of like 1 or something if we don't already